app/src/main/java/com/nervousfish/nervousfish/activities/SharedActivityMethods.java

```diff
@@ -0,0 +1,143 @@
+package com.nervousfish.nervousfish.activities;
+
+import android.app.Activity;
+import android.app.AlertDialog;
+import android.content.DialogInterface;
+import android.widget.EditText;
+
+import com.nervousfish.nervousfish.R;
+import com.nervousfish.nervousfish.data_objects.Contact;
+import com.nervousfish.nervousfish.modules.database.IDatabase;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.List;
+
+import cn.pedant.SweetAlert.SweetAlertDialog;
+
+/**
+ * Contains shared functionality between classes to reduce code duplication
+ */
+// Suppressed because we have an empty enum
+@SuppressWarnings("checkstyle:nowhitespacebefore")
+enum SharedActivityMethods {
+    ;
+    private static final Logger LOGGER = LoggerFactory.getLogger(SharedActivityMethods.class);
+
+    /**
+     * Adds a new contact to the database. If it existed already, it asks the user if he wants to add
+     * the received public key to the existing contact, or if he wants to create a contact with a new name.
+     *
+     * @param database The database in which the contact should be stored
+     * @param activity The activity that should draw the prompts
+     * @param contact  The newly received contact
+     */
+    static void newContactReceived(final IDatabase database, final Activity activity, final Contact contact) {
+        if (checkExists(database, contact)) {
+            LOGGER.warn("Contact already existed...");
+            handleExistingContact(database, activity, contact);
+        } else {
+            LOGGER.info("Adding contact to database...");
+            try {
+                database.addContact(contact);
+            } catch (final IOException e) {
+                LOGGER.error("Could not add contact to databae", e);
+            }
+        }
+    }
+
+    private static void handleExistingContact(final IDatabase database, final Activity activity, final Contact contact) {
+        new SweetAlertDialog(activity, SweetAlertDialog.WARNING_TYPE)
+                .setTitleText(activity.getString(R.string.contact_already_exists))
+                .setContentText(String.format(activity.getString(R.string.contact_exists), contact.getName()))
+                .setConfirmText(activity.getString(R.string.add_public_key_to_contact))
+                .setCancelText(activity.getString(R.string.create_new_contact))
+                .showCancelButton(true)
+                .setCancelClickListener(new SharedActivityMethods.CreateNewContactClickListener(activity, database, contact))
+                .setConfirmClickListener(new SharedActivityMethods.AddPublicKeyToContactsClickListener(database, contact))
+                .show();
```

Comment:
I don't think so, because if I want to save a Word file with the same name as an existing file, then it asks me if I want to rename it. If I do that I won't get a notification afterwards that successfully a new file with the name I entered is created